### PR TITLE
Applying golangci-lint

### PR DESCRIPTION
**Pull request by junghao**

### CODE REVIEW

File: cmd/fdsn-quake-consumer/sc3ml.go [hunk 1]
```diff
@@ -203,10 +203,11 @@ func (e *event) save() error {
 
 	_, err = txn.Exec(`DELETE FROM fdsn.event WHERE PublicID = $1 AND ModificationTime <= $2`, e.PublicID, e.ModificationTime)
 	if err != nil {
-		txn.Rollback()
+		_ = txn.Rollback()
```
Comment (bpeng) [inline cmd/fdsn-quake-consumer/sc3ml.go]: Shall this error be handled/logged?

File: cmd/fdsn-ws-nrt/data_nrt_test.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func BenchmarkGetRecordDB(b *testing.B) {
 }
 
 // funcs for setting up test data.
-
+//nolint:deadcode
```
Comment (danieldooley) [inline cmd/fdsn-ws-nrt/data_nrt_test.go]: Please add reasons to all your exclusions. If you put a second comment it won't affect the linter. i.e:
```
//nolint:deadcode //[reason]
```

File: cmd/fdsn-quake-consumer/sc3ml.go [hunk 3]
```diff
@@ -222,14 +226,20 @@ func (e *event) save() error {
...
+					log.Println(e)
+				}
 				return nil
 			} else {
-				txn.Rollback()
+				if e := txn.Rollback(); e != nil {
+					log.Println(e)
+				}
 				return err
 			}
```
Comment (danieldooley) [inline cmd/fdsn-quake-consumer/sc3ml.go]: Would this be cleaner?
```
if errorUnique.Code == `23505` {
  err = nil
}
if e := txn.Rollback(); e != nil {
  log.Println(e)
}
return err

File: cmd/fdsn-quake-consumer/sc3ml.go [hunk 4]
```diff
@@ -203,10 +204,13 @@ func (e *event) save() error {
 
 	_, err = txn.Exec(`DELETE FROM fdsn.event WHERE PublicID = $1 AND ModificationTime <= $2`, e.PublicID, e.ModificationTime)
 	if err != nil {
-		txn.Rollback()
+		if e := txn.Rollback(); e != nil {
+			log.Println(e)
```
Comment (danieldooley) [inline cmd/fdsn-quake-consumer/sc3ml.go]: Some context would be good here: `Rollback Failed: %v` or similar

File: cmd/fdsn-ws-nrt/data_nrt.go [hunk 5]
```diff
@@ -19,7 +19,9 @@ var errNoData = errors.New("no data")
 // start and end should be set for all queries.
 func holdingsSearchNrt(d fdsn.DataSearch) ([]string, error) {
 	timer := metrics.Start()
-	defer timer.Track("holdingsSearchNrt")
+	defer func() {
+		_ = timer.Track("holdingsSearchNrt")
```
Comment (danieldooley) [inline cmd/fdsn-ws-nrt/data_nrt.go]: probably just chuck a log in here

File: cmd/fdsn-ws/fdsn_station.go [hunk 6]
```diff
@@ -142,8 +142,11 @@ func init() {
 		log.Println("Loading fdsn station xml file ", "etc/"+s3Meta)
 		var f *os.File
 		if f, err = os.Open("etc/" + s3Meta); err == nil {
-			io.Copy(by, f)
-			f.Close()
+			_, err = io.Copy(by, f)
+			if err != nil {
+				log.Println("Error copying station xml file", err)
+			}
+			_ = f.Close()
```
Comment (danieldooley) [inline cmd/fdsn-ws/fdsn_station.go]: Wouldn't hurt to log here

File: cmd/fdsn-quake-consumer/sc3ml.go [hunk 7]
```diff
@@ -203,10 +204,13 @@ func (e *event) save() error {
...
 	if err != nil {
-		txn.Rollback()
+		if e := txn.Rollback(); e != nil {
+			log.Printf("Rollback Failed: %v", e)
+		}
 		return err
 	}
 
+	//nolint:gosec	// Unavoidable string concatenation
 	_, err = txn.Exec(`INSERT INTO fdsn.event(`+strings.TrimSuffix(insert, ", ")+`) VALUES( `+strings.TrimSuffix(params, ", ")+` )`, values...)
```
Comment (danieldooley) [inline cmd/fdsn-quake-consumer/sc3ml.go]: Would a `fmt.Sprintf` trigger a linter warning here as well?

I understand that this is safe, but it concerns me that we have a //nolint:gosec next to an `sql` function.

Could you try creating the string on another line (maybe with `fmt.Sprintf` for better readability) and passing it in as a variable?

Comment (junghao) [reply cmd/fdsn-quake-consumer/sc3ml.go]: Tried with fmt.Sprintf still got the warning of 
```cmd/fdsn-quake-consumer/sc3ml.go:213: G201: SQL string formatting (gosec)```

Comment (danieldooley) [reply cmd/fdsn-quake-consumer/sc3ml.go]: What if you assign the string to a variable and pass it into `txn.Exec` like that?

Comment (junghao) [reply cmd/fdsn-quake-consumer/sc3ml.go]: Yes that's what I did.
```
	q := fmt.Sprintf("INSERT INTO fdsn.event(%s) VALUES(%s)", strings.TrimSuffix(insert, ", "), strings.TrimSuffix(params, ", "))
	_, err = txn.Exec(q, values...)
```
The code above still being rejected.
But the error is at the line of fmt . 
I think the linter checks context to see whether a string formatting is for SQL execution.

Similar static analysis and workarounds happened here:
https://github.com/GeoNet/haz/pull/220/files#diff-c32caf3c852145c15537fee84e4b625aR57

Comment (junghao) [reply cmd/fdsn-quake-consumer/sc3ml.go]: I'll expand the "INSERT INTO" to resolve the lint error.

Comment (junghao) [reply cmd/fdsn-quake-consumer/sc3ml.go]: @danieldooley I've updated the PR. Could you please check again? Thanks.

**Pull request merged**